### ISSUE TITLE
Having another LSP alongside null-ls sources makes it so only one source is reported in the virtual text.

### ISSUE THREAD

**Issue by vinicius507** [jose-elias-alvarez/null-ls.nvim#181]
### FAQ

- [X] I have checked [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

0.5.0

### Steps to reproduce

1. Register some LSP diagnostics source other than `null-ls`(e.g. clangd with lspconfig) and a source from `null-ls` for the same filetype.
2. Edit a file of said filetype and trigger errors for both sources on the same line.

### Expected behavior

The prefix of the LSP diagnostics should show the count of diagnostics from all sources.

### Actual behavior

Null-ls diagnostics aren't counted on the prefix.

Screenshot of the virtual text and `trouble.nvim`:
![image](https://user-images.githubusercontent.com/6608056/132881044-e866d5b4-bc5a-4b17-bba2-677b0bbdedf6.png)

### Debug log

```
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received LSP request for method initialize
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received LSP notification for method initialized
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received LSP notification for method textDocument/didOpen
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: running generators for method NULL_LS_DIAGNOSTICS
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: spawning command /home/vinicius/.local/bin/norminette with args:
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: { "--filename", "/tmp/teste.c", "--cfile", "/* ************************************************************************** */\n/*                                                                            */\n/*          :::      ::::::::   */\n/*   teste.c                                            :+:      :+:    :+:   */\n/*                                                    +:+ +:+      +:+     */\n/*   By: vgocalv <vgocalv@student.42sp.org.br>      +#+  +:+       +#+        */\n/*                                                +#+#+#+#+#+   +#+           */\n/*   Created: 2021/09/10 12:30:09 by vgocalv           #+#    #+#             */\n/*   Updated: 2021/09/10 12:30:09 by vgocalv          ###   ########.fr       */\n/*                                                                */\n/* ************************************************************************** */\n\n#include <stdio.h>\n\nint\tmain(oid)\n{\n\tchar *str;\n\n\t(void)str;\n\treturn (0);\n}\n" }
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: error output: /tmp/teste.c: Error!
Error: MISSING_IDENTIFIER   (line:  15, col:  14):	missing type qualifier or identifier in function arguments
Error: SPACE_REPLACE_TAB    (line:  17, col:   9):	Found space when expecting tab

[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: output: nil
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received diagnostics from generators
[DEBUG Fri 10 Sep 2021 12:43:22 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: { {
    col = "14",
    message = "NORME: Missing type qualifier or identifier in function arguments(MISSING_IDENTIFIER)",
    range = {
      end = {
        character = 0,
        line = 15
      },
      start = {
        character = 13,
        line = 14
      }
    },
    row = "15",
    severity = 1,
    source = "norminette"
  }, {
    col = "9",
    message = "NORME: Found space when expecting tab(SPACE_REPLACE_TAB)",
    range = {
      end = {
        character = 0,
        line = 17
      },
      start = {
        character = 8,
        line = 16
      }
    },
    row = "17",
    severity = 1,
    source = "norminette"
  } }
[DEBUG Fri 10 Sep 2021 12:43:23 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received LSP request for method shutdown
[DEBUG Fri 10 Sep 2021 12:43:23 PM -03] .../site/pack/packer/opt/null-ls.nvim/lua/null-ls/utils.lua:51: received LSP notification for method exit
```

### Help

Yes, but I don't know how to start. I would need guidance

**Comment by jose-elias-alvarez** [comment]
Hmm, I don't use the `virtualtext` option myself, but are you using a custom handler? When using the default handler, this is what I see in the case you describe (the first error is from null-ls, and the second is from `tsserver`):

<img width="1126" alt="Screen Shot 2021-09-17 at 9 13 29 AM" src="https://user-images.githubusercontent.com/54108223/133788590-82043e9b-8b86-46dd-be5e-353bba29647b.png">

**Comment by vinicius507** [comment]
Indeed, I'm using a custom handler:

```lua
vim.lsp.handlers['textDocument/publishDiagnostics'] = vim.lsp.with(
	vim.lsp.diagnostic.on_publish_diagnostics,
	{
		virtual_text = {
			spacing = 4,
			prefix = '●',
		},
		signs = true,
		underline = true,
		update_in_insert = true,
	}
)
```
I'll try it out with the default handler ;)

**Comment by vinicius507** [comment]
The problem persists with the default handler. I don't know why, though.

![image](https://user-images.githubusercontent.com/6608056/133797708-19edf599-6b03-4563-b4b8-a7900d97efc7.png)

**Comment by jose-elias-alvarez** [comment]
I see, could you try to put together a reproduction with specific sources and a sample file?

**Comment by vinicius507** [comment]
> I see, could you try to put together a reproduction with specific sources and a sample file?

Would a typescript example suffice? Will try to reproduce it now.

**Comment by jose-elias-alvarez** [comment]
Sure, TypeScript would be ideal.

**Comment by vinicius507** [comment]
[Repository for the example](https://github.com/vinicius507/null-ls-virtualtext-bug-example)

I've created a sample project using `npx create-next-app --typescript` with `create-next-app v11.0.1`.
I've introduced an error in the file `pages/index.tsx` in which I call `test(alo)`, a non existing function with a non existing parameter.

The virtual text shows 2 errors:
![image](https://user-images.githubusercontent.com/6608056/133801336-369460d0-bdcc-4096-bd4e-2eef9d33c950.png)

Yet `trouble.nvim` displays 4 errors:
![image](https://user-images.githubusercontent.com/6608056/133801403-0a590bf9-127a-45fb-81ab-90b38ff8bca9.png)

**Comment by jose-elias-alvarez** [comment]
Sorry it took me so long to get around to this. I believe this issue has been solved in 0.6.0. On the latest master (`NVIM v0.6.0-dev+575-g2ef9d2a66`) this is what I see with your config:

<img width="648" alt="Screen Shot 2021-11-11 at 6 55 03 PM" src="https://user-images.githubusercontent.com/54108223/141385648-6fe320c8-c4dd-4dbe-a35c-8fb5b7c0ded3.png">

It looks like they're now grouped by source, which perhaps isn't ideal but is an improvement. If you're still on 0.5.0 / 0.5.1, I think the best solution would be to write a custom handler or wait for the 0.6.0 release.

**Comment by vinicius507** [comment]
> Sorry it took me so long to get around to this. I believe this issue has been solved in 0.6.0. On the latest master (`NVIM v0.6.0-dev+575-g2ef9d2a66`) this is what I see with your config:
> 
> <img alt="Screen Shot 2021-11-11 at 6 55 03 PM" width="648" src="https://user-images.githubusercontent.com/54108223/141385648-6fe320c8-c4dd-4dbe-a35c-8fb5b7c0ded3.png">
> 
> It looks like they're now grouped by source, which perhaps isn't ideal but is an improvement. If you're still on 0.5.0 / 0.5.1, I think the best solution would be to write a custom handler or wait for the 0.6.0 release.

That was exactly what I did. Thanks :)
